tests/unit/test_auto_regression.py

```diff
@@ -593,3 +593,175 @@ def test_fit_auto_regression_np(lags):
+@pytest.mark.parametrize("intercept", [1.0, -4.0])
+def test_fit_autoregression_monthly_np(slope, intercept):
+    # test if autoregrerssion can fit using previous month as independent variable
+    # and current month as dependent variable
+    # NOTE: the fit for the slope is bounded between -1 and 1
+    np.random.seed(0)
+    prev_month = np.random.normal(size=100)
+    cur_month = prev_month * slope + intercept
+
+    result = mesmer.stats._auto_regression._fit_auto_regression_monthly_np(
+        cur_month, prev_month
+    )
+    expected = (slope, intercept)
+    np.testing.assert_allclose(result, expected)
+
+
+@pytest.mark.parametrize("slope", [2.0, -2])
+def test_fit_autoregression_monthly_np_outside_bounds(slope):
+    # test that given a slope outside the bounds of -1 and 1, the slope is clipped
+    np.random.seed(0)
+    prev_month = np.random.normal(size=100)
+    cur_month = prev_month * slope + 3
+
+    result = mesmer.stats._auto_regression._fit_auto_regression_monthly_np(
+        cur_month, prev_month
+    )
+    expected_slope = 1.0 * np.sign(slope)
+    np.testing.assert_allclose(result[0], expected_slope)
+
+
+def test_fit_auto_regression_monthly():
+    n_years = 20
+    n_gridcells = 10
+    np.random.seed(0)
+    data = xr.DataArray(
+        np.random.normal(size=(n_years * 12, n_gridcells)),
+        dims=("time", "gridcell"),
+        coords={
+            "time": pd.date_range("2000-01-01", periods=n_years * 12, freq="M"),
+            "gridcell": np.arange(n_gridcells),
+        },
+    )
+
+    result = mesmer.stats.fit_auto_regression_monthly(data)
+
+    _check_dataset_form(result, "result", required_vars={"slope", "intercept"})
+    _check_dataarray_form(
+        result.slope,
+        "slope",
+        ndim=2,
+        required_dims={"month", "gridcell"},
+        shape=(12, n_gridcells),
+    )
+    _check_dataarray_form(
+        result.intercept,
+        "intercept",
+        ndim=2,
+        required_dims={"month", "gridcell"},
+        shape=(12, n_gridcells),
+    )
+
+    with pytest.raises(TypeError, match="Expected a `xr.DataArray`"):
+        mesmer.stats.fit_auto_regression_monthly(data.values)
+
+
+def test_predict_auto_regression_monthly_np():
+    slope = np.repeat(0.0, 12)
+    intercept = np.arange(1, 13)
+    result = mesmer.stats._auto_regression._predict_auto_regression_monthly_np(
+        intercept, slope, 120, 0
+    )
+
+    np.testing.assert_allclose(result, np.tile(intercept, 10).reshape(10, 12))
+
+    with pytest.raises(
+        ValueError, match="The number of time steps must be a multiple of 12"
+    ):
+        mesmer.stats._auto_regression._predict_auto_regression_monthly_np(
+            intercept, slope, 100, 0
+        )
+
+
+@pytest.mark.parametrize("buffer", [2, 5, 10])
+def test_predict_auto_regression_monthly_np_buffer(buffer):
+    slope = np.random.normal(size=12)
+    intercept = np.ones(12)
+    n_ts = 120
+    result_wo_buffer = (
```

Comment:
```suggestion
    res_wo_buffer = (
```
Bikeshedding but avoiding the annoying wrap